### ISSUE TITLE
How to build test samples?

### ISSUE THREAD

**Issue by dizcza** [cpc/hipcl#28]
I've successfully compiled and built clang+LLVM, LLVM-SPIRV Translator, and HIPCL as in a Dockerfile I've described in https://github.com/cpc/hipcl/issues/11.

I'm trying to compile HIPCL test samples inside a docker container.

```
cd samples
mkdir testsbuild
cd testsbuild
LLVM_INSTALL_DIR=/root/llvm/build cmake -DCMAKE_CXX_COMPILER=${LLVM_INSTALL_DIR}/bin/clang++ -DCMAKE_C_COMPILER=${LLVM_INSTALL_DIR}/bin/clang ..
```

I get a bunch of errors
```
-- The C compiler identification is Clang 8.0.1
-- The CXX compiler identification is Clang 8.0.1
-- Check for working C compiler: /root/llvm/build/bin/clang
-- Check for working C compiler: /root/llvm/build/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /root/llvm/build/bin/clang++
-- Check for working CXX compiler: /root/llvm/build/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:25 (install):
  install TARGETS given no RUNTIME DESTINATION for executable target
  "hipmath".
Call Stack (most recent call first):
  hipmath/CMakeLists.txt:2 (add_hipcl_test)

...

CMake Error at CMakeLists.txt:25 (install):
  install TARGETS given no RUNTIME DESTINATION for executable target
  "SimpleConvolution".
Call Stack (most recent call first):
  hip-cuda/SimpleConvolution/CMakeLists.txt:3 (add_hipcl_test)


CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.10)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.
```

I also tried to compile a single source file (let's call it `asset.hip`) and got another error:

```
$ /root/hipcl/build/bin/hipcl_config -C
-D__HIP_PLATFORM_HIPCL__=  -x hip --hip-device-lib-path=/root/hipcl/build/share  --hip-device-lib=kernellib.bc --hip-llvm-pass-path=/root/hipcl/build/lib/llvm  -I/root/hipcl/build/include
$ /root/llvm/build/bin/clang++ -pthread -fPIE -O2 -g -std=c++11 `/root/hipcl/build/bin/hipcl_config -C`  -o asset.o asset.hip -Wl,-rpath,/root/hipcl/build/lib -L/root/hipcl/build/lib -lhipcl

clang-8: error: no such file or directory: 'kernellib.bc'
```

**Comment by franz** [comment]
Hello,

this error

`clang-8: error: no such file or directory: 'kernellib.bc'`

means clang can't find kernellib.bc. The location of the file depends on whether you are building your example (`asset.hip`) against an installed hipcl, or a hipcl build tree. `hipcl_config` returns the options/flags for an installed hipcl, so unless you have already built & installed it, those options are not usable. The hipcl examples are built against hipcl build tree, so the options are different.

Can you pastebin the Dockerfile somewhere ?

**Comment by dizcza** [comment]
You're right, I've not run `make install` - silly of me.

After `make install`, I ran VecAdd test but

```
/root/hipcl/build/bin/samples/VecAdd: error while loading shared libraries: libhipcl.so.0: cannot open shared object file: No such file or directory
```

I symlinked `ln -s /root/hipcl/build/lib/libhipcl.so.0 /usr/local/lib/libhipcl.so.0` but still was getting this error.

Then I ran `ldconfig` and this resolved the issue. Perhaps, it's worth mentioning `ldconfig` in README how to build HIPCL and run the tests.

Sadly, I cannot build my asset.hip file because it relies on `hipMemcpyToSymbol` function which, as README states, is not supported. Hope you'll add the support soon.

Thanks for the project. Looking forward to finally run my example on a built-in laptop Intel GPU.
